### PR TITLE
feat/rbac roles

### PR DESCRIPTION
**Pull request by ykhedher**

### CODE REVIEW

File: src/component/common/CreateConfirm/CreateConfirm.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            clearTimeout(timeout);
+        };
+    }, []);
+
+    const styles = useStyles();
+    const confettiColors = ['#d13447', '#ffbf00', '#263672'];
+    const confettiAmount = 200;
+
+    const getRandomNumber = (input: number) => {
+        return Math.floor(Math.random() * input) + 1;
```
[the remark this answers is not in the document]
Comment (FredrikOseberg) [reply src/component/common/CreateConfirm/CreateConfirm.tsx:34]: ignore

File: src/component/admin/ProjectRoles/RolesList.tsx [hunk 2]
```diff
@@ -12,20 +12,53 @@ import { ADMIN } from '../../providers/AccessProvider/permissions';
...
+        try {
+            await deleteRole(currentRole.id);
+            refetch();
+            setToastData({
+                show: true,
+                type: 'success',
+                text: 'Successfully deleted role.',
+            });
+        } catch (e) {
+            console.log(e);
```
Comment (FredrikOseberg) [inline src/component/admin/ProjectRoles/RolesList.tsx:47]: Set toast data here if it fails.

File: src/component/admin/ProjectRoles/RolesListItem/RoleListItem.styles.ts [hunk 3]
```diff
@@ -6,8 +6,11 @@ export const useStyles = makeStyles(theme => ({
...
     },
-    leftTableCell:{
+    leftTableCell: {
         textAlign: 'left',
-        maxWidth: '300px'
-    }
+        maxWidth: '300px',
+    },
+    icon: {
+        color: '#0000008a',
```
Comment (FredrikOseberg) [inline src/component/admin/ProjectRoles/RolesListItem/RoleListItem.styles.ts:14]: theme.palette.grey here

File: src/component/admin/api-token/ApiTokenList/ApiTokenList.tsx [hunk 4]
```diff
@@ -80,19 +80,25 @@ const ApiTokenList = ({ location }: IApiTokenList) => {
...
+            setToastApiError(e.message);
+        }
     };
     const copyToken = (value: string) => {
         if (copy(value)) {
             setToastData({
                 type: 'success',
-                show: true,
+                title: 'Token copied',
+                confetti: true,
```
Comment (chriswk) [inline src/component/admin/api-token/ApiTokenList/ApiTokenList.tsx:101]: :confetti_ball:

**Review by chriswk — APPROVED** [review]
This looks good, couldn't even find a nit on spellings :+1:  :smile:

**Pull request merged**